addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1088,18 +1094,29 @@ var SnippetEditor = Widget.extend({
         // Remove the siblings that belong to a snippet in grid mode
         // and put the identified grid mode snippets in their own "selector".
         const selectorGrids = new Set();
-        if (rowEl.classList.contains('row')) {
-            if ($selectorSiblings) {
-                // Looping backwards because elements are removed, so the
-                // indexes are not lost.
-                for (let i = $selectorSiblings.length - 1; i >= 0; i--) {
-                    if (isPopup && !$selectorSiblings[i].closest('div.s_popup')) {
-                        // Removing the siblings that are outside the popup if
-                        // the grid item is in a popup.
-                        $selectorSiblings.splice(i, 1);
-                    } else {
-                        const gridSnippet = $selectorSiblings[i].closest('div.o_grid_mode');
-                        if (gridSnippet) {
+        if (rowEl.classList.contains('row') && $selectorSiblings) {
+            // Looping backwards because elements are removed, so the
+            // indexes are not lost.
+            for (let i = $selectorSiblings.length - 1; i >= 0; i--) {
+                if (isPopup && !$selectorSiblings[i].closest('div.s_popup')) {
+                    // Removing the siblings that are outside the popup if
+                    // the grid item is in a popup.
+                    $selectorSiblings.splice(i, 1);
+                } else {
+                    const gridSnippet = $selectorSiblings[i].closest('div.o_grid_mode');
+                    if (gridSnippet) {
+                        // The grid in which the dragged column is (if any).
+                        const targetGrid = rowEl.classList.contains('o_grid_mode') || rowEl.closest('div.o_grid_mode');
+                        if (!hasGridLayoutOption) {
+                            if (gridSnippet === $selectorSiblings[i].parentElement && gridSnippet === targetGrid) {
```

Comment:
I am a bit lost, in which case do you need to find `$selectorSiblings[i].closest('div.o_grid_mode')` (as you define `gridSnippet`) searching for something else than its direct parent?

Reply:
Before this commit, it is true that I could have looked at the parent directly instead, since when you drag columns the only dropzones appearing are direct children of the row.

But it is not the case anymore if you have inner snippets: for example, when you drag a form field, which is a column, dropzones appear in the form (between the fields) and in the "outer" snippet (between the columns). For the outer ones, then the grid is the parent but for the inner ones, we need to find the closest grid to know that we are in a grid. So looking for the closest combines both cases. 
Note that this case happens because there are nested rows, which I generally avoided by blocking the grid mode for the concerned snippets, but I forgot the fact that Form could be dropped in a column.

Reply:
Note 2: I forgot to mention it but normally, after the PR https://github.com/odoo/odoo/pull/94029 will be merged, there won't be any dropzones appearing outside the forms so this part of the fix may not be necessary anymore (as only form was concerned by this issue).

Reply:
I r+'ed that PR :) Once it is forward-ported, maybe review this code and ping me again once you think what is to be kept has been kept :+1: